lib/web/device_trust.go

```diff
@@ -76,7 +81,29 @@ func (h *Handler) deviceWebConfirm(w http.ResponseWriter, r *http.Request, _ htt
 
 	// Always redirect back to the dashboard, regardless of outcome.
 	app.SetRedirectPageHeaders(w.Header(), "" /* nonce */)
-	http.Redirect(w, r, "/web", http.StatusSeeOther)
+
+	redirectTo := "/web"
+
+	if redirectParam != "" {
+		parsedURL, err := url.Parse(redirectParam)
+		if err == nil {
+			cleanPath := path.Clean(parsedURL.Path)
```

Comment:
This block seems like it could be a small helper function of its own.

Reply:
Added a helper and tests